db/csv.go

```diff
@@ -0,0 +1,118 @@
+package db
+
+import (
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	warehouse "github.com/diegocmsantos/warehouse/core"
+)
+
+type LineValidator struct {
+	Validators []func([]string) (bool, error)
```

Comment:
Seems like you were anticipating validation rules a little too much-- premature optimization is the root of all evil and here we complicate signatures anticipating multiple validators, when we only have one defined in the problem